lib/inspec/secrets/yaml.rb

```diff
@@ -24,12 +24,18 @@ def initialize(target)
         @inputs = ::YAML.load_file(target)
       end
 
-      if @inputs == false || !@inputs.is_a?(Hash)
-        Inspec::Log.warn("#{self.class} unable to parse #{target}: invalid YAML or contents is not a Hash")
-        @inputs = nil
+      # In case of empty yaml file raise the warning else raise the parsing error.
+      if !@inputs || @inputs.empty?
+        Inspec::Log.warn("Unable to parse #{target}: YAML file is empty.")
+        @inputs = nil if @inputs == false
```

Comment:
Small refactoring suggestion, instead of `if @inputs == false` we can use `unless @inputs`. Just the way you have refactored in `if` condition

Reply:
This is only in case of input is false cause input will be already nil if its nil but I will add similar checks here too. Thanks!!

Reply:
That check itself is not required. Removing it.